src/core/algorithm/boundingbox.cpp

```diff
@@ -0,0 +1,194 @@
+        return buff.first.type == BufferType::PositionAttrib;
+    });
+    if (it != buffers.end() && it->second->getSize() > 0) {
+
+        const auto minmax =
+            it->second->getRepresentation<BufferRAM>()->dispatch<std::pair<dvec4, dvec4>>(
+                [](auto br) {
+                    using ValueType = util::PrecisionValueType<decltype(br)>;
+                    const auto &data = br->getDataContainer();
+
+                    using Res = std::pair<ValueType, ValueType>;
+                    Res minmax{DataFormat<ValueType>::max(), DataFormat<ValueType>::lowest()};
+                    minmax =
+                        std::accumulate(data.begin(), data.end(), minmax,
+                                        [](const Res &mm, const ValueType &v) -> Res {
+                                            return {glm::min(mm.first, v), glm::max(mm.second, v)};
+                                        });
+
+                    return std::pair<dvec4, dvec4>{util::glm_convert<dvec4>(minmax.first),
+                                                   util::glm_convert<dvec4>(minmax.second)};
+                });
+
+        mat4 trans = mesh.getCoordinateTransformer().getDataToWorldMatrix();
+        worldMin = glm::min(worldMin, vec3(trans * vec4(vec3(minmax.first), 1.f)));
+        worldMax = glm::max(worldMax, vec3(trans * vec4(vec3(minmax.second), 1.f)));
+    } else {
+        // No vertices, use same values for min/max
+        worldMin = worldMax = mesh.getOffset();
+    }
+
+    auto m = glm::scale(worldMax - worldMin);
+    m[3] = vec4(worldMin, 1.0f);
+    return m;
+}
+
+mat4 boundingBox(const std::vector<std::shared_ptr<const Mesh>> &meshes) {
+    if (meshes.empty()) return mat4(0.f);
+
+    vec3 worldMin(std::numeric_limits<float>::max());
+    vec3 worldMax(std::numeric_limits<float>::lowest());
+    for (const auto &mesh : meshes) {
+        auto bb = boundingBox(*mesh);
+        worldMin = glm::min(worldMin, vec3{bb * vec4{0.0f, 0.0f, 0.0f, 1.0f}});
+        worldMax = glm::max(worldMax, vec3{bb * vec4{1.0f, 1.0f, 1.0f, 1.0f}});
+    }
+    auto m = glm::scale(worldMax - worldMin);
+    m[3] = vec4(worldMin, 1.0f);
+    return m;
+}
+
+mat4 boundingBox(const Volume &volume) {
+    return volume.getCoordinateTransformer().getDataToWorldMatrix();
+}
+
+mat4 boundingBox(const std::vector<std::shared_ptr<Volume>> &volumes) {
+    if (volumes.empty()) return mat4(0.f);
+
+    vec3 worldMin(std::numeric_limits<float>::max());
+    vec3 worldMax(std::numeric_limits<float>::lowest());
+    for (const auto &volume : volumes) {
+        auto bb = boundingBox(*volume);
+        worldMin = glm::min(worldMin, vec3{bb * vec4{0.0f, 0.0f, 0.0f, 1.0f}});
+        worldMax = glm::max(worldMax, vec3{bb * vec4{1.0f, 1.0f, 1.0f, 1.0f}});
+    }
+    auto m = glm::scale(worldMax - worldMin);
+    m[3] = vec4(worldMin, 1.0f);
+    return m;
+}
+
+std::function<std::optional<mat4>()> boundingBox(const DataInport<Mesh> &mesh) {
```

Comment:
I suppose this could be templated to support all MeshPorts as: 
```c++
template<unsigned N, bool Flat>
std::function<std::optional<mat4>()> boundingBox(const DataInport<Mesh,N,Flat> &mesh) {
    return [port = &mesh]() -> std::optional<mat4> {
        if (port->hasData()) {
            if constexpr (N == 1){
                return boundingBox(*port->getData());
            } else {
                return boundingBox(port->getVectorData());
            }
        } else {
            return std::nullopt;
        }
    };
}
```

Reply:
There are overloads for the different versions we use, regular multi and flat. Wanted to avoid some includes... but forward decls was difficult